pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -288,6 +300,28 @@ func enrichClusterSystemData(doc *api.OpenShiftClusterDocument, systemData *api.
 	}
 }
 
+func persistIdentityUrl(cluster *api.OpenShiftCluster, identityURL []string, isCreate bool) error {
```

Comment:
nit: since this function isn't actually persisting data, perhaps it could be renamed `validateIdentityURL` and return the valid `identityURL` to then be set on `doc.OpenShiftCluster` instead of passing it in as a side effect.

Reply:
I tried this, but it reduces the scope of the unit test coverage, since we're no longer testing that the correct identity URL was set in the cluster document and only testing that the identity URL was validated correctly.